network/monitor_linux.go

```diff
@@ -34,13 +38,32 @@ func (nm *networkManager) AddStateRulesToMap() map[string]string {
 		snatKey := fmt.Sprintf("-s Unicast -o %s -j snat --to-src %s --snat-arp --snat-target ACCEPT", extIf.Name, extIf.MacAddress.String())
 		rulesMap[snatKey] = ebtables.PostRouting
 
+		for _, extIP := range extIf.IPAddresses {
+			if extIP.IP.To4() != nil {
+				arpReplyKey := fmt.Sprintf("-p ARP --arp-op Request --arp-ip-dst %s -j arpreply --arpreply-mac %s", extIP.IP.String(), extIf.MacAddress.String())
+				rulesMap[arpReplyKey] = ebtables.PreRouting
+			}
+		}
+
 		for _, nw := range extIf.Networks {
 			for _, ep := range nw.Endpoints {
 				for _, ipAddr := range ep.IPAddresses {
-					arpReplyKey := fmt.Sprintf("-p ARP --arp-op Request --arp-ip-dst %s -j arpreply --arpreply-mac %s", ipAddr.IP.String(), ep.MacAddress.String())
-					rulesMap[arpReplyKey] = ebtables.PreRouting
+					if ipAddr.IP.To4() != nil {
+						arpReplyKey := fmt.Sprintf("-p ARP --arp-op Request --arp-ip-dst %s -j arpreply --arpreply-mac %s", ipAddr.IP.String(), ep.MacAddress.String())
+						rulesMap[arpReplyKey] = ebtables.PreRouting
+					}
+
+					dst := "--ip-dst"
+					proto := "IPv4"
+					ipAddress := ipAddr.IP.String()
+					if ipAddr.IP.To4() == nil {
+						dst = "--ip6-dst"
+						proto = "IPv6"
+						ipAddress = ipAddr.IP.String() + ipv6Mask
+					}
 
-					dnatMacKey := fmt.Sprintf("-p IPv4 -i %s --ip-dst %s -j dnat --to-dst %s --dnat-target ACCEPT", extIf.Name, ipAddr.IP.String(), ep.MacAddress.String())
```

Comment:
Do you need to do something similar for arpDnatKey above?

Reply:
nope not needed..there is no arp responder for ipv6

Reply:
Then lgtm